src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -308,6 +320,269 @@ private PKCS8EncodedKeySpec getKeySpecImpl(Key decryptKey,
+    public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key,
+        char[] password, String algorithm, AlgorithmParameterSpec params,
+        Provider provider) {
+
+        PBEKeySpec keySpec = new PBEKeySpec(password);
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(algorithm);
+        SecretKey skey;
+
+        try {
+            SecretKeyFactory factory;
+            if (provider == null) {
+                factory = SecretKeyFactory.getInstance(algorithm);
+            } else {
+                factory = SecretKeyFactory.getInstance(algorithm, provider);
+            }
+            skey = factory.generateSecret(keySpec);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new IllegalArgumentException(e);
+        }
+        return encryptKeyImpl(key, algorithm, skey, params, provider, null);
+    }
+
+    /**
+     * Creates and encrypts an {@code EncryptedPrivateKeyInfo} from a given
+     * {@code PrivateKey} and password.  Default algorithm and parameters are
+     * used.
+     *
+     * @param key The {@code PrivateKey} object to encrypt.
+     * @param password the password used in the PBE encryption.  This array is
+     *                 cloned before being used.
+     * @return an {@code EncryptedPrivateKeyInfo}.
+     * @throws IllegalArgumentException on initialization errors based on the
+     * arguments passed to the method.
+     * @throws SecurityException on a encryption errors.
+     * @throws NullPointerException when the password is null.
+     *
+     * @implNote The `jdk.epkcs8.defaultAlgorithm` Security Property defines
+     * the default encryption algorithm and the {@code AlgorithmParameterSpec}
+     * are the provider's algorithm defaults.
+     *
+     * @since 25
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.PEM_API)
+    public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key,
+        char[] password) {
+        char[] pass = password.clone();
+        if (Pem.DEFAULT_ALGO == null || Pem.DEFAULT_ALGO.length() == 0) {
+            throw new SecurityException("Security property " +
+                "\"jdk.epkcs8.defaultAlgorithm\" may not specify a " +
+                "valid algorithm.  Operation cannot be performed.");
+        }
+        return encryptKey(key, pass, Pem.DEFAULT_ALGO,
+            null, null);
+    }
+
+    /**
+     * Creates and encrypts an {@code EncryptedPrivateKeyInfo} from a given
+     * {@link PrivateKey} using the {@code encKey} and given parameters.
+     *
+     * If {@code algorithm} is {@code null} the default algorithm will be used.
+     * {@code params} is null, the provider default will be used.
+     * {@code params} is {@code null}.  The {@code provider} or {@code random} will
+     * be selected through the default provider list if set to {@code null}.
+     *
+     * @param key the {@code PrivateKey} object to encrypt.
+     * @param encKey the encryption {@code Key}
```

Comment:
It will be more precise if we say `key` is the key to be encrypted and `encKey` is the key used to encrypt `key`.